woocommerce/woocommerce-ios
# [Bundle products in order form] Show a configuration screen with v0 design

Part of #10428 


⚠️ Please review https://github.com/woocommerce/woocommerce-ios/pull/10990 first

## Why


To break down the changes, this PR just contains the views & view models for the new bundle product configure screen. The configuration submission function is a no-op now, and will be added in a future PR.

## How

The configuration screen for the bundle product is `ConfigurableBundleProductView` with view model `ConfigurableBundleProductViewModel` - the main logic in the view model is to load all the products for the bundle items. After all the products are loaded, then each bundle item & its product are used to render a new view `ConfigurableBundleItemView`. The configurable properties for each bundle item:

- Quantity
- (If the bundle item is optional) Whether the bundle item is included
- (If the bundle item is variable) Selected variation (full support will be implemented in a separate subtask)

Due to the large number of diffs already, I'll leave unit tests for a future subtask.

## Testing instructions


Prerequisite: the store has [Product Bundles extension](https://woocommerce.com/products/product-bundles/) enabled, and also a bundle product with non-empty items **including a variable product**.

- Go to the Orders tab
- Tap `+` to create an order, or tap on an existing order then tap `Edit`
- Tap `Add Products`
- Search and select a bundle product in the prerequisite --> the bundle product should be added to the order form
- Tap on the expand  🔽 CTA
- Tap `Configure` --> after a bit (products being loaded remotely), the configurable bundle items should be shown similar to the menu in core
- For the bundle item that is variable, tap `Select variation` --> a list of allowed variations should be shown
- Select a variation --> the attributes should be shown on the Configure screen
- Tap `Done` --> the Configure modal should be dismissed without any actions

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/348f46bf-32dc-4fe9-8ae3-a83e78beb09b" width="300" />


---
